airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/file_type_parser.py

```diff
@@ -19,14 +20,16 @@ class FileTypeParser(ABC):
     """
 
     @abstractmethod
-    async def infer_schema(self, file: RemoteFile, stream_reader: AbstractFileBasedStreamReader) -> Schema:
+    async def infer_schema(self, config: FileBasedStreamConfig, file: RemoteFile, stream_reader: AbstractFileBasedStreamReader) -> Schema:
```

Comment:
we could do that. the main argument for is that we don't know how other formats might use the config object so this is just a little more flexible. Do you think that would be valid since you've looked more into the other formats?

Reply:
It feels like they should only need info from the format-specific part of the dictionary but I'm fine with keeping it more flexible until we know for sure.

Reply:
Actually I'm realizing that I have a need for a different part of the config in a separate PR that I'm working on, so yeah, let's keep it as you've got it.